src/app/icd/IcdMonitoringTable.h

```diff
@@ -29,25 +31,54 @@ inline constexpr size_t kIcdMonitoringBufferSize = 40;
 
 struct IcdMonitoringEntry : public PersistentData<kIcdMonitoringBufferSize>
 {
-    static constexpr size_t kKeyMaxSize = 16;
 
     IcdMonitoringEntry(FabricIndex fabric = kUndefinedFabricIndex, NodeId nodeId = kUndefinedNodeId)
     {
         this->fabricIndex      = fabric;
         this->checkInNodeID    = nodeId;
         this->monitoredSubject = nodeId;
     }
+
+    IcdMonitoringEntry(Crypto::SessionKeystore * keyStore, FabricIndex fabric = kUndefinedFabricIndex,
```

Comment:
It's sort of weird that we need to pass in a keystore even if we just plan to use this as an outparam for a get from a table that has a keystore.... but OK.